src/network/clientpackethandler.cpp

```diff
@@ -1803,20 +1803,32 @@ void Client::handleCommand_SetLighting(NetworkPacket *pkt)
 {
 	Lighting& lighting = m_env.getLocalPlayer()->getLighting();
 
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.shadow_intensity;
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.saturation;
-	if (pkt->getRemainingBytes() >= 24) {
-		*pkt >> lighting.exposure.luminance_min
-				>> lighting.exposure.luminance_max
-				>> lighting.exposure.exposure_correction
-				>> lighting.exposure.speed_dark_bright
-				>> lighting.exposure.speed_bright_dark
-				>> lighting.exposure.center_weight_power;
-	}
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.volumetric_light_strength;
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.shadow_tint;
+	if (pkt->getRemainingBytes() < 4)
+		return;
+	*pkt >> lighting.shadow_intensity;
+
+	if (pkt->getRemainingBytes() < 4)
+		return;
+	*pkt >> lighting.saturation;
+
+	if (pkt->getRemainingBytes() < 24)
+		return;
+	*pkt >> lighting.exposure.luminance_min
+			>> lighting.exposure.luminance_max
+			>> lighting.exposure.exposure_correction
+			>> lighting.exposure.speed_dark_bright
+			>> lighting.exposure.speed_bright_dark
+			>> lighting.exposure.center_weight_power;
+
+	if (pkt->getRemainingBytes() < 4)
+		return;
+	*pkt >> lighting.volumetric_light_strength;
+
+	if (pkt->getRemainingBytes() < 4)
+		return;
+	*pkt >> lighting.ambient_light;
```

Comment:
Rebase mistake? `shadow_tint` should come before this, `ambient_light` should be last for backwards compat

Reply:
Probably yes...  There were multiple of conflicts and it was necessary to fix them almost in all 18 commits.